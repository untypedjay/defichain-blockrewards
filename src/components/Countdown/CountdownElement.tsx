import React from 'react';
import { StyledContent, StyledLabel } from '../../styles/CardStyle';

interface Props {
  children: number;
  label: string;
}

export default function CountdownElement({ children, label }: Props) {
  return (
    <div>
      <StyledContent>{ children }</StyledContent>
      <StyledLabel>{ label }</StyledLabel>
    </div>
  );
}