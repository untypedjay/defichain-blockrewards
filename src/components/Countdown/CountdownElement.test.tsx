import ReactDOM from 'react-dom';
import React from 'react';
import CountdownElement from './CountdownElement';
import { cleanup, render } from '@testing-library/react';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CountdownElement label="days">{ 3 }</CountdownElement>, div);
});

it('renders correctly', () => {
  render(<CountdownElement label="days">{ 3 }</CountdownElement>);
});

it('matches snapshot', () => {
  const { asFragment } = render(<CountdownElement label="days">{ 3 }</CountdownElement>);
  expect(asFragment()).toMatchSnapshot();
});