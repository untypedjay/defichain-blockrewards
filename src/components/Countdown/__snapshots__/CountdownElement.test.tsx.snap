// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p
      class="sc-bdfBwQ dHtfKI"
    >
      3
    </p>
    <p
      class="sc-gsTCUz fEJTeQ"
    >
      days
    </p>
  </div>
</DocumentFragment>
`;
